presto-hive/src/test/java/com/facebook/presto/hive/s3/security/TestAWSS3SecurityMapping.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.s3.security;
+
+import com.facebook.presto.cache.CacheConfig;
+import com.facebook.presto.hive.DynamicConfigurationProvider;
+import com.facebook.presto.hive.HdfsContext;
+import com.facebook.presto.hive.HiveClientConfig;
+import com.facebook.presto.hive.HiveSessionProperties;
+import com.facebook.presto.hive.OrcFileWriterConfig;
+import com.facebook.presto.hive.ParquetFileWriterConfig;
+import com.facebook.presto.hive.aws.security.AWSSecurityMappingConfig;
+import com.facebook.presto.hive.aws.security.AWSSecurityMappingsSupplier;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.security.AccessDeniedException;
+import com.facebook.presto.spi.security.ConnectorIdentity;
+import com.facebook.presto.testing.TestingConnectorSession;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Optional;
+
+import static com.facebook.presto.hive.s3.S3ConfigurationUpdater.S3_ACCESS_KEY;
+import static com.facebook.presto.hive.s3.S3ConfigurationUpdater.S3_IAM_ROLE;
+import static com.facebook.presto.hive.s3.S3ConfigurationUpdater.S3_SECRET_KEY;
+import static com.facebook.presto.hive.s3.security.TestAWSS3SecurityMapping.MappingResult.credentials;
+import static com.facebook.presto.hive.s3.security.TestAWSS3SecurityMapping.MappingResult.role;
+import static com.facebook.presto.hive.s3.security.TestAWSS3SecurityMapping.MappingSelector.empty;
+import static com.google.common.io.Resources.getResource;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
```

Comment:
Personal preference, but I try to stick to a single test framework for simplicity. Mixing and matching is confusing